### PR TITLE
Circle CI changes: go build and test

### PR DESCRIPTION
**Pull request by drinkcoffee**
Run go build first, then run the tests. 
Run tests with docker containers / blockchains running.

Signed-off-by: Peter Robinson <drinkcoffee@eml.cc>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -262,24 +262,80 @@ jobs:
...
+            echo; git -P reflog --date=iso
+            go version
+            pwd
+            ls -al
+            echo GOROOT: $GOROOT
+      - run:
+          name: Upgrade golang
+          command: |
+            sudo rm -rf /usr/local/go
+            curl -O https://dl.google.com/go/go1.17.1.linux-amd64.tar.gz
```
Comment (ermyas) [inline .circleci/config.yml:285]: nit: should we use the same go version as the version used by the docker image that runs the tests? i.e. 1.17.3?

File: .circleci/config.yml [hunk 2]
```diff
@@ -262,24 +262,80 @@ jobs:
           name: Build production code
           command: make build
           working_directory: ~/project/messaging/relayer
+
+  relayer-test:
+    machine:
+      image: ubuntu-2004:202107-02
```
Comment (ermyas) [inline .circleci/config.yml:268]: should we instead use the latest machine image `ubuntu-2004:202111-01`? [It seems to already have Go 1.17.3 installed](https://discuss.circleci.com/t/linux-machine-executor-images-2021-november-q4-update/42005), which might remove the need for the upgrade step below

**Review by ermyas — APPROVED** [review]
LGTM, minor comments added

File: .circleci/config.yml (hunk 1 above)
Comment (drinkcoffee) [reply .circleci/config.yml]: This will be fixed by addressing the comment below.

File: .circleci/config.yml (hunk 2 above)
Comment (drinkcoffee) [reply .circleci/config.yml]: Good point. Done.

**Pull request merged**